ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -92,24 +101,20 @@ void handleRestoration(final RestoreCommands restoreCommands) {
           }
         }
     );
+    ksqlEngine.getPersistentQueries().forEach(PersistentQueryMetadata::start);
```

Comment:
So this is the main change. We delay execution in `restore` mode until we process all the statements first.